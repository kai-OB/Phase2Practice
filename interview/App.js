/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Button,
  Text,
  View,
} from 'react-native';
/* $FlowFixMe[missing-local-annot] The type annotation(s) required by Flow's
 * LTI update could not be added via codemod */
import InputFields from './Components/InputFields/content'
import Result from './Components/Result/content'

function App () {

  return (
    <SafeAreaView>
      <Text>Morgage Calculator</Text>
        <View>
         <InputFields
          title={'Desired Loan Amount'}
          value = {'2'}/>
           <InputFields
          title={'Estimated APR'}
          value = {'2'}/>
          <InputFields
          title={'Desired Term'}
          value = {'2'}/>
          {/* <Button
            onPress={}>
          </Button> */}
          <Result/>
        </View>
    </SafeAreaView>
  );
};
export default App;
