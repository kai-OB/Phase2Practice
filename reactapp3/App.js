/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import Node from 'react';
import Section from 'Components/Section';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';


const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <Section
        title= 'To Do List:'
        children='Hopefully with authentication and backend'>
      </Section>
    </SafeAreaView>
  );
};

export default App;
