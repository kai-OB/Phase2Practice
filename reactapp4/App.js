/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
//import { NativeRouter, Route, Link } from "react-router-native";
import { StyleSheet, View} from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
// import { createStackNavigator } from '@react-navigation/stack';
import { createStackNavigator} from 'reactapp4/node_modules/@react-navigation/native-stack';
import HomeScreen from './Components/HomeScreen/content';
 const Stack = createStackNavigator(blank,blank);
 

function App() {

  return (
   
    <NavigationContainer>
       <HomeScreen/>
       <Stack.Navigator>
         <Stack.Screen name="Home" component={HomeScreen} />
       </Stack.Navigator>
     </NavigationContainer>
  );
}
export default App;