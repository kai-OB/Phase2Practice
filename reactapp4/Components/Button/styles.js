import { StyleSheet,useColorScheme } from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';
//const isDarkMode =useColorScheme() === 'dark';

const styles = StyleSheet.create({
  
  container:{
   width:'60%',
   marginLeft: '20%',
  },
  buttonStyle: {
    backgroundColor: Colors.darker,
    height: 40,
    borderRadius: 20, //radius half of height (rounds corners)
    alignItems: 'center',
    justifyContent:'center',
   // backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  },
  buttonTitle: {
    //color: isDarkMode ? Colors.black : Colors.white,
    fontSize:85,
    color: 'white',
  }
});

export default styles;
