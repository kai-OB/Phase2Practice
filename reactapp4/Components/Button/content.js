/*
* Section is a section of text
* An optional title and an optional Description
*/
import React from 'react';
import {Text, View, useColorScheme,Pressable} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import styles from './styles';

const Button = (props) =>{    // will take a primary or a secondary type
  // const type = props.type;
  const content = props.content;
  const onPress = props.onPress;
  // since they have the same name, this is the same as
  //const { type, content, onPress} = props;

  return(
      <View style = {styles.container}>
          <Pressable 
            style={styles.buttonStyle}
            onPress={()=>{ onPress()}}>
              <Text style = {[styles.buttonText,{color:Colors.lighter}]}>{content}</Text>
          </Pressable>
      </View>
  );
};
export default Button;