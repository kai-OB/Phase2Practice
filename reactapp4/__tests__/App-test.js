/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Button from '../Components/Button/content';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
afterEach(cleanup);

// it('renders correctly', () => {
//   renderer.create(<App />);
// });
// test('Button renders correctly + Snapshor', () => {
//   const tree = renderer.create(< />).toJSON();
//   expect(tree).toMatchSnapshot();
//   console.log(tree);
// });
