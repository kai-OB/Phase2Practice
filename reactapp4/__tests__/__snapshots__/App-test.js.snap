// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly part2 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            },
            Object {
              "backgroundColor": "#F3F3F3",
            },
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../Libraries/NewAppScreen/components/logo.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "marginBottom": -192,
                "marginLeft": -128,
                "opacity": 0.2,
                "overflow": "visible",
                "resizeMode": "cover",
              },
            ]
          }
          testID="new-app-screen-header"
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 40,
                "fontWeight": "700",
                "textAlign": "center",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Welcome to
          

          React Native
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Step One
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                Object {
                  "color": "#444",
                },
              ]
            }
          >
            Edit 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              App.js
            </Text>
             to change this screen and then come back to see your edits.
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
